liquidprompt

```diff
@@ -1594,6 +1595,23 @@ _lp_as_text()
     echo -nE "$1" | sed -$_LP_SED_EXTENDED "s,$_LP_CLEAN_ESC,,g"
 }
 
+# Get length of a given string, after evaluating escape codes (\u for user, etc)
+_lp_length()
+{
+    typeset len evaluated_ps1
+    typeset printable_ps1=$(_lp_as_text "$1")
+
+    if $_LP_SHELL_bash; then
+        evaluated_ps1=$(printf '%s' "${printable_ps1@P}")
+    else # zsh
+        evaluated_ps1=$(print -P $printable_ps1)
+    fi
+
+    len=${#evaluated_ps1}
+
+    echo $len
```

Comment:
Post-rework Liquidprompt avoids using stdout to return values. Use a variable, named after the function:
```suggestion
    lp_prompt_length=${#evaluated_ps1}
```
Oh and name the function `__lp_prompt_length()`. The name is too generic, and lies about what it does.